spec/lib/ruby_smb/smb2/packet/transform_header_spec.rb

```diff
@@ -0,0 +1,220 @@
+    let(:struct) { RubySMB::SMB2::Packet::TreeConnectRequest.new }
+
+    it 'raises the expected exception if the given algorithm is invalid' do
+      expect { packet.encrypt(struct, key, algorithm: 'RC4') }.to raise_error(
+        RubySMB::Error::EncryptionError,
+        'Error while encrypting with \'RC4\' (ArgumentError: Invalid algorithm, must be either AES-128-CCM or AES-128-GCM)'
+      )
+    end
+
+    context 'with AES-128-GCM algorithm (default)' do
+      before :example do
+        begin
+          OpenSSL::Cipher.new('AES-128-GCM')
+        rescue
+          skip(
+            "This test cannot be run since the version of OpenSSL the ruby "\
+            "OpenSSL extension was built with (#{OpenSSL::OPENSSL_VERSION}) "\
+            "does not support AES-128-GCM cipher")
+        end
+      end
+
+      it 'generates a cipher using OpenSSL::Cipher' do
+        expect(OpenSSL::Cipher).to receive(:new).with('AES-128-GCM').and_call_original
+        packet.encrypt(struct, key)
+      end
+
+      it 'encrypts a BinData structure' do
+        packet.encrypt(struct, key)
+        expect(packet.decrypt(key)).to eq(struct.to_binary_s)
+      end
+
+      it 'encrypts a string' do
+        packet.encrypt('data', key)
+        expect(packet.decrypt(key)).to eq('data')
+      end
+
+      it 'raises the expected exception if an error occurs' do
+        allow(OpenSSL::Cipher).to receive(:new).and_raise(
+          RuntimeError.new('unsupported cipher algorithm (AES-128-GCM)'))
+        expect { packet.encrypt('data', key) }.to raise_error(
+          RubySMB::Error::EncryptionError,
+          'Error while encrypting with \'AES-128-GCM\' (RuntimeError: unsupported cipher algorithm (AES-128-GCM))'
+        )
+      end
+    end
+
+    context 'with AES-128-CCM algorithm' do
+      it 'generates a cipher using OpenSSL::CCM' do
+        expect(OpenSSL::CCM).to receive(:new).with('AES', key, 16).and_call_original
+        packet.encrypt(struct, key, algorithm: 'AES-128-CCM')
+      end
+
+      it 'encrypts a BinData structure' do
+        packet.encrypt(struct, key, algorithm: 'AES-128-CCM')
+        expect(packet.decrypt(key, algorithm: 'AES-128-CCM')).to eq(struct.to_binary_s)
+      end
+
+      it 'encrypts a string' do
+        packet.encrypt('data', key, algorithm: 'AES-128-CCM')
+        expect(packet.decrypt(key, algorithm: 'AES-128-CCM')).to eq('data')
+      end
+
+      it 'raises the expected exception if an error occurs' do
+        allow(OpenSSL::CCM).to receive(:new).and_raise(
+          OpenSSL::CCMError.new('unsupported cipher algorithm (AES-128-CCM)'))
+        expect { packet.encrypt('data', key, algorithm: 'AES-128-CCM') }.to raise_error(
+          RubySMB::Error::EncryptionError,
+          'Error while encrypting with \'AES-128-CCM\' (OpenSSL::CCMError: unsupported cipher algorithm (AES-128-CCM))'
+        )
+      end
+    end
+  end
+
+  it 'reads a binary data as expected' do
```

Comment:
Minor grammatical change:
```suggestion
  it 'reads binary data as expected' do
```